mmcv/cnn/bricks/upsample.py

```diff
@@ -65,20 +69,11 @@ def build_upsample_layer(cfg, *args, **kwargs):
     Returns:
         nn.Module: Created upsample layer.
     """
-    if not isinstance(cfg, dict):
-        raise TypeError(f'cfg must be a dict, but got {type(cfg)}')
-    if 'type' not in cfg:
-        raise KeyError(
-            f'the cfg dict must contain the key "type", but got {cfg}')
-    cfg_ = cfg.copy()
-
-    layer_type = cfg_.pop('type')
-    if layer_type not in UPSAMPLE_LAYERS:
-        raise KeyError(f'Unrecognized upsample type {layer_type}')
-    else:
-        upsample = UPSAMPLE_LAYERS.get(layer_type)
-
-    if upsample is nn.Upsample:
-        cfg_['mode'] = layer_type
-    layer = upsample(*args, **kwargs, **cfg_)
-    return layer
+
+    warnings.warn(
+        ImportWarning(
+            '``build_upsample_layer(cfg, *args, **kwargs)`` will be '
+            'deprecated. Please use '
+            '``UPSAMPLE_LAYERS.build(cfg, *args, **kwargs)`` instead.'))
+
+    return build_upsample_layer_from_cfg(cfg, UPSAMPLE_LAYERS, *args, **kwargs)
```

Comment:
Should use UPSAMPLE_LAYERS.buil() here?

Reply:
Will lead to recursive import.

Reply:
Then please leave a comment above this line of code.